cooltools/lib/numutils.py

```diff
@@ -141,6 +141,64 @@ def fill_nainf(arr, value=0, copy=True):
     arr[~np.isfinite(arr)] = value
     return arr
 
+def interp_nan(a_init, pad_zeros=True, verbose=False):
+    '''linearly interpolate to fill nans in a vector or a matrix
+    
+    Parameters
+    ----------
+
+    a_init : np.array
+
+    pad_zeros : bool, optional
+        If True, pads the matrix with zeros to fill nans at the edges.
+        By default, True.
+    
+    Returns
+    -------
+    interp : flattened np.array with nans linearly interpolated
+
+    Notes
+    -----
+    adapted from:
+        https://stackoverflow.com/a/37882746   
+        https://stackoverflow.com/a/39592604
+    '''
+
+    init_shape = np.shape(a_init)
+    if len(init_shape) == 2 and init_shape[0] != 1 and init_shape[1] !=1:
+        if verbose==True: print('interpolating 2D matrix')
+        if pad_zeros == True:
+            a = np.zeros((init_shape[0]+2,init_shape[1]+2))
+            a[1:-1,1:-1] = a_init
+        else:
+            a = a_init
+        x, y = np.indices(a.shape)
+        interp = np.array(a)
+        interp[np.isnan(interp)] = scipy.interpolate.griddata(
+                 (x[~np.isnan(a)], y[~np.isnan(a)]), # points we know
```

Comment:
Pre-computing the `~np.isnan(a)` mask could speed things up a bit.

Reply:
It should even be `np.isnan(interp)`

Reply:
replaced with a much faster method based on scipy.interpolate.RegularGridInterpolator